### PR TITLE
feat: new option for auto-populating image credits

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

#2858 added behavior to automatically populate image credits if a `credit` field existed on the image's EXIF or IPTC metadata. Some publishers do not want this behavior. This PR adds an option to theme settings to disable this behavior to accommodate those publishers.

Closes `1206709674365921/1206754105927437` .

### How to test the changes in this Pull Request:

1. Check out this branch.
2. Visit **Newspack > Site Design > Settings**. Observe a new setting under "Media Credits" to "Auto-populate image credits", enabled by default

Screenshot 2024-03-05 at 12 11 53 AM

3. Go to **Media Library > Add New Media File** and upload an image that contains credit EXIF data (see Asana task for an example image for testing).
4. Confirm that the "Credit" field is automatically populated from the credit EXIF field.
5. Toggle off the "Auto-populate image credits" in **Newspack > Site Design > Settings**.
6. Repeat step 3 and confirm that the credit is NOT auto-populated.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206754105927437

### CODE REVIEW

**Comment by gamebits** [comment]
My understanding of the Slack convo is that this feature should be opt-in, and the toggle should be off by default.

File: includes/class-newspack-image-credits.php [hunk 1]
```diff
@@ -369,7 +371,26 @@ public static function get_default_settings() {
 				'type'        => 'image',
 				'value'       => null,
 			],
+			[
+				'description' => __( 'Automatically populate image credits from EXIF or IPTC metadata when uploading new images.', 'newspack-image-credits' ),
+				'key'         => 'newspack_image_credits_auto_populate',
+				'label'       => __( 'Auto-populate image credits', 'newspack-image-credits' ),
+				'type'        => 'boolean',
+				'value'       => true,
```
Comment (chickenn00dle) [inline includes/class-newspack-image-credits.php]: Looks like this should be false per this comment: https://github.com/Automattic/newspack-plugin/pull/2973#issuecomment-1978779652

File: assets/wizards/site-design/views/theme-settings/index.js [hunk 2]
```diff
@@ -35,6 +35,7 @@ const ThemeSettings = props => {
 		newspack_image_credits_placeholder_url: imageCreditsPlaceholderUrl,
 		newspack_image_credits_class_name: imageCreditsClassName = '',
 		newspack_image_credits_prefix_label: imageCreditsPrefix = '',
+		newspack_image_credits_auto_populate: imageCreditsAutoPopulate = true,
```
Comment (chickenn00dle) [inline assets/wizards/site-design/views/theme-settings/index.js]: This should also default to false per this comment: https://github.com/Automattic/newspack-plugin/pull/2973#issuecomment-1978779652

**Review by chickenn00dle — CHANGES_REQUESTED** [review]
Nice! This works as expected and is pretty much good to go once the default value for this new option is set to false.

File: assets/wizards/site-design/views/theme-settings/index.js (hunk 2 above)
Comment (dkoo) [reply assets/wizards/site-design/views/theme-settings/index.js]: Updated in 91dd7bed

File: includes/class-newspack-image-credits.php (hunk 1 above)
Comment (dkoo) [reply includes/class-newspack-image-credits.php]: Updated in 91dd7bed

**Review by chickenn00dle — APPROVED** [review]

**Pull request merged**